feat: Replace `http(s)` with `node-fetch`

shrink opened the pull request:
Closes #82

Replaces `http(s)` with `node-fetch`. The public-facing API of the library is unchanged but the mechanism for communicating with the API has changed and could manifest in different behaviour, e.g: previously we were handling errors in different places with different logic. There's also an increase in the minimum node version from 10 to 14.

Added a Workflow to test the library with each supported version of node.

- [ ] Release as v4

max-ipinfo commented on __tests__/ipinfoWrapper.test.ts:
```diff
@@ -184,4 +184,9 @@ describe("IPinfoWrapper", () => {
+        const ipinfo = new IPinfoWrapper("invalid-token");
+        await expect(ipinfo.lookupIp("1.2.3.4")).rejects.toThrow();
+    });
 });
```
Could we add a unit test that simulates a JSON parse error?

max-ipinfo commented:
@shrink Feel free to include a semver upgrade to 4.x.x 

Here is an example from the last release: https://github.com/ipinfo/node/commit/089af589027d0662d85675a2e1f4a0ba22ef4058

shrink commented:
@max-ipinfo added the changelog entry and new version. I've also simplified the documentation and added information about catching errors as a best practice. The documentation includes separate TypeScript and JavaScript examples but as TypeScript can run all JavaScript we do not need separate examples: I have consolidated the examples.

max-ipinfo reviewed (approved):
LGTM!
